pkg/assembler/backends/ent/backend/vulnEqual.go

```diff
@@ -16,139 +16,309 @@
 )
 
 func (b *EntBackend) VulnEqual(ctx context.Context, filter *model.VulnEqualSpec) ([]*model.VulnEqual, error) {
 
-	var where = []predicate.VulnEqual{
+	if len(filter.Vulnerabilities) > 2 {
+		return nil, fmt.Errorf("too many vulnerability specified in vuln equal filter")
+	}
+
+	query := b.client.VulnEqual.Query().
+		Where(vulnEqualQuery(filter)).
+		WithVulnerabilityA(func(query *ent.VulnerabilityIDQuery) {}).
+		WithVulnerabilityB(func(query *ent.VulnerabilityIDQuery) {})
+	results, err := query.Limit(MaxPageSize).All(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return collect(results, toModelVulnEqual), nil
+}
+
+func vulnEqualQuery(filter *model.VulnEqualSpec) predicate.VulnEqual {
+	if filter == nil {
+		return NoOpSelector()
+	}
+	where := []predicate.VulnEqual{
 		optionalPredicate(filter.ID, IDEQ),
 		optionalPredicate(filter.Justification, vulnequal.JustificationEQ),
 		optionalPredicate(filter.Origin, vulnequal.OriginEQ),
 		optionalPredicate(filter.Collector, vulnequal.CollectorEQ),
 	}
-	for _, vulnID := range filter.Vulnerabilities {
-		where = append(where, vulnequal.HasVulnerabilityIdsWith(optionalPredicate(vulnID.VulnerabilityID, vulnerabilityid.VulnerabilityIDEqualFold)))
-		where = append(where, vulnequal.HasVulnerabilityIdsWith(vulnerabilityid.HasTypeWith(optionalPredicate(vulnID.Type, vulnerabilitytype.TypeEqualFold))))
-		if vulnID.NoVuln != nil {
-			if *vulnID.NoVuln {
-				where = append(where, vulnequal.HasVulnerabilityIdsWith(vulnerabilityid.HasTypeWith(vulnerabilitytype.TypeEqualFold(NoVuln))))
+
+	if len(filter.Vulnerabilities) == 1 {
+		where = append(where, vulnequal.Or(vulnequal.HasVulnerabilityAWith(optionalPredicate(filter.Vulnerabilities[0].VulnerabilityID, vulnerabilityid.VulnerabilityIDEqualFold), optionalPredicate(filter.Vulnerabilities[0].Type, vulnerabilityid.TypeEqualFold)),
+			vulnequal.HasVulnerabilityBWith(optionalPredicate(filter.Vulnerabilities[0].VulnerabilityID, vulnerabilityid.VulnerabilityIDEqualFold), optionalPredicate(filter.Vulnerabilities[0].Type, vulnerabilityid.TypeEqualFold))))
+
+		if filter.Vulnerabilities[0].NoVuln != nil {
+			if *filter.Vulnerabilities[0].NoVuln {
+				where = append(where, vulnequal.Or(vulnequal.HasVulnerabilityAWith(vulnerabilityid.TypeEqualFold(NoVuln)), vulnequal.HasVulnerabilityBWith(vulnerabilityid.TypeEqualFold(NoVuln))))
 			} else {
-				where = append(where, vulnequal.HasVulnerabilityIdsWith(vulnerabilityid.HasTypeWith(vulnerabilitytype.TypeNEQ(NoVuln))))
+				where = append(where, vulnequal.Or(vulnequal.HasVulnerabilityAWith(vulnerabilityid.TypeNEQ(NoVuln)), vulnequal.HasVulnerabilityBWith(vulnerabilityid.TypeNEQ(NoVuln))))
 			}
 		}
-	}
+	} else if len(filter.Vulnerabilities) == 2 {
+		where = append(where, vulnequal.Or(vulnequal.HasVulnerabilityAWith(optionalPredicate(filter.Vulnerabilities[0].VulnerabilityID, vulnerabilityid.VulnerabilityIDEqualFold), optionalPredicate(filter.Vulnerabilities[0].Type, vulnerabilityid.TypeEqualFold)),
```

Comment:
```suggestion
		where = append(where, vulnequal.Or(vulnequal.HasVulnerabilityAWith(optionalPredicate(filter.Vulnerabilities[0].ID, IDEQ), optionalPredicate(filter.Vulnerabilities[0].VulnerabilityID, vulnerabilityid.VulnerabilityIDEqualFold), optionalPredicate(filter.Vulnerabilities[0].Type, vulnerabilityid.TypeEqualFold)),
```

Reply:
ah yes the ID check. We will need to work on fixing the queries for the tests.